--- vgabios.c	2006-07-10 09:47:51.000000000 +0200
+++ vgabios.c	2007-03-12 17:08:06.000000000 +0100
@@ -149,16 +149,35 @@
            
   jmp vgabios_init_func
 
-vgabios_name:
-.ascii	"Plex86/Bochs VGABios"
-.ascii	" "
-.byte	0x00
+.byte 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
+
+.word pci_data
 
 // Info from Bart Oldeman
 .org 0x1e
 .ascii  "IBM"
 .byte   0x00
 
+vgabios_name:
+.ascii	"Plex86/Bochs VGABios"
+.ascii	" "
+.byte	0x00
+
+
+// PCI data structure by Stefan Reinauer
+pci_data:
+.ascii "PCIR"		// signature
+.word 0x1013, 0x00b8	// vendor, device
+.word 0x0000		// vpd
+.word 0x0018		// pci data structure len
+.byte 0x00		// pci data structure revision
+.byte 0x00, 0x00, 0x03	// class code
+.word 0x46		// image length
+.word 0x0001		// image revision
+.byte 0x00		// code type: x86
+.byte 0x80		// flags: last image in rom
+.word 0x0000		// reserved
+
 vgabios_version:
 #ifndef VGABIOS_VERS
 .ascii	"current-cvs"
